src/AdminBuilder.js

```diff
@@ -1,38 +1,62 @@
-import React, {Component} from 'react';
-import PropTypes from 'prop-types';
-import {Admin, Resource, Delete} from 'admin-on-rest';
 import Api from '@api-platform/api-doc-parser/lib/Api';
-import List from './List';
-import Show from './Show';
+import {Admin, Delete, Resource} from 'admin-on-rest';
+import PropTypes from 'prop-types';
+import React from 'react';
 import Create from './Create';
 import Edit from './Edit';
+import fieldFactory from './fieldFactory';
+import inputFactory from './inputFactory';
+import List from './List';
+import Show from './Show';
 
-export default class extends Component {
-  static propTypes = {
-    api: PropTypes.instanceOf(Api).isRequired,
-    restClient: PropTypes.func.isRequired,
-  };
-
-  render() {
-    let props = {...this.props};
-    if (!props.title) props.title = this.props.api.title;
+const AdminBuilder = props => {
+  const {api, fieldFactory, inputFactory, title = api.title} = props;
 
-    return (
-      <Admin {...props}>
-        {this.props.api.resources.map(resource => (
+  return (
+    <Admin {...props} title={title}>
+      {api.resources.map(resource => {
+        const {
+          create = Create,
+          edit = Edit,
+          list = List,
+          name,
+          props,
+          remove = Delete,
+          show = Show,
+        } = resource;
+        return (
           <Resource
-            options={{api: this.props.api, resource}}
-            key={resource.name}
-            name={resource.name}
-            list={List}
-            show={Show}
-            create={Create}
-            edit={Edit}
-            remove={Delete}
-            {...resource.props}
+            {...props}
```

Comment:
I'm not sure but I think you can
```
```
{
  ...props, 
  create, 
  edit,
  remove,
  show,
  list,
  name
}

Reply:
It does not work but the following syntax works:

```js
            {...{
              ...props,
              create,
              edit,
              key: name,
              list,
              name,
              options: {
                api,
                fieldFactory,
                inputFactory,
                resource,
              },
              remove,
              show,
            }}
```

Do you prefer this syntax?

Reply:
nope ^^

Reply:
I think it’s better than the above as it’s more explicit, and therefore more readable.